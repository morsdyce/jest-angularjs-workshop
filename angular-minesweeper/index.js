import 'common/assets/stylesheets/minefield.css';

import App from 'angular-minesweeper/components/app/app.module';
import Common from 'angular-minesweeper/common/common.module';
import Board from 'angular-minesweeper/components/board/board.module';

export default angular.module('ngReact', [
  Common.name,
  Board.name,
  App.name
]);