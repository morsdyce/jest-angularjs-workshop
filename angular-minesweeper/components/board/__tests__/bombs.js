import 'angular-minesweeper';

describe('<bombs>', () => {
  let createComponent, $scope, element;

  beforeEach(angular.mock.module('ngReact.board'));

  beforeEach(inject(($rootScope, $compile) => {
    createComponent = (html, scopeAttrs) => {
      $scope = angular.extend($rootScope.$new(), scopeAttrs);

      element = angular.element(html);
      element = $compile(element)($scope);
      $scope.$digest();
    };
  }));

  test('should display 5 bombs', () => {
    const html = `<bombs amount="amount"></bombs>`;
    createComponent(html, { amount: 5 });

    expect(element.html()).toContain('5');
    expect(element.html()).toMatchSnapshot();
  });

});

