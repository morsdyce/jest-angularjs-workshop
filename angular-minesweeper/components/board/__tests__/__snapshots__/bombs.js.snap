// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bombs> should display 5 bombs 1`] = `
"<div class=\\"bomb-count number-window\\">
  <span class=\\"ng-binding\\">
    5
  </span>
</div>



"
`;
